Chore/add check sprint day max slot

riomar0001 opened the pull request.

Commit:
feat(payment): implement registration data extraction and email notification for payment failures, enhancing error handling in payment processing

Commit:
feat(registration): enhance registration handling by returning existing registration data instead of conflict response

Commit:
feat(payment): add handling for failed and cancelled payment notifications, including fetching existing registrations and enhancing email templates for user engagement

Commit:
feat(permissions): add DynamoDB index permissions for paymentHandler to enhance data access

ASPactores committed:
feat(email): add endpoint to resend confirmation email for registrations

ASPactores committed:
feat(discount): implement function to adjust max usage of reusable discount codes

ASPactores committed:
feat(payment): enhance payment event processing to skip duplicate registrations based on email

ASPactores committed:
feat(email): implement resend confirmation email functionality for event registrations

ASPactores committed:
refactor(email): remove resend confirmation email endpoint and associated logic

riomar0001 committed:
chore: create login for sprint day max slots

jlorion committed:
modify logic for sprintday guard clause

jlorion committed:
feat(sprintday): added sprintday attributes on events, track sprintday registrations, update event to have sprintday max slots

riomar0001 committed:
Merge branch 'main' of github.com:DurianPy-Davao-Python-User-Group/TechTix into chore/add-check-sprint-day-max-slot

riomar0001 committed:
Merge branch 'chore/add-check-sprint-day-max-slot' of github.com:DurianPy-Davao-Python-User-Group/TechTix into chore/add-check-sprint-day-max-slot

riomar0001 committed:
feat: update add event form with limiting sprint day

riomar0001 committed:
fix: map isSprintDayLimitedSlot set true if there is maximumSprintDaySlots

riomar0001 committed:
chore: implement error handling for updating if new max slot for sprint is less than the registered in sprint

riomar0001 committed:
feat: disable ticket availability for sprint if sold

jlorion committed:
fix(events): update registration error on empty registration

riomar0001 committed:
UI: remove price and change text to sold out when sprint day slot is full

riomar0001 committed:
Merge branch 'chore/add-check-sprint-day-max-slot' of github.com:DurianPy-Davao-Python-User-Group/TechTix into chore/add-check-sprint-day-max-slot

riomar0001 committed:
chore: make sold out for easy to indentify

riomar0001 committed:
format: run prettier

riomar0001 committed:
fix: rever original file for dummy_ammpliy_outputs

riomar0001 committed:
format: run prettier

riomar0001 committed:
chore: make SOLD OUt more easy to identify

jlorion committed:
fix: sprint ticket is false on default if max slots

jlorion committed:
fix: remove unused import

jlorion committed:
fix event dto lacking attributes

seangaaab commented on frontend/dummy_amplify_outputs copy.json:
rename this back to `dummy_amplify_outputs.json`

seangaaab commented on frontend/src/pages/client/pycon/register/steps/TicketSelectionStep.tsx:
```diff
@@ -226,10 +232,17 @@ const TicketType: FC<TicketTypeProps> = ({ ticketType, benefits, subtitle, value
-const SprintDaySection: FC<SprintDaySectionProps> = ({ isSelected, sprintDayPrice, onChange }) => {
+const SprintDaySection: FC<SprintDaySectionProps> = ({ maximumSprintDaySlots, sprintDayRegistrationCount, isSelected, sprintDayPrice, onChange }) => {
+  const sprintIsSoldOut = maximumSprintDaySlots === sprintDayRegistrationCount;
+  if (sprintIsSoldOut) {
```
move this inside a `useEffect`. Add `sprintIsSoldOut` in the `useEffect`'s dependency array.

riomar0001 replied:
Hello Sean, I'm not sure on what you mean to add sprintIsSoldOut inside the useEffect.

should I create a state for this?

seangaaab commented on frontend/src/pages/client/pycon/register/steps/TicketSelectionStep.tsx:
```diff
@@ -292,27 +315,40 @@ const SprintDaySection: FC<SprintDaySectionProps> = ({ isSelected, sprintDayPric
-            </Button>
+            {sprintIsSoldOut ? (
+              <div className=" rounded-lg p-4 text-center">
+                <h3 className="font-nunito font-extrabold text-4xl text-pycon-red mb-2">SOLD OUT!</h3>
```
The `Sold out!` word should be where the price (+300) is located. It should be inside price container

<img width="819" height="409" alt="image" src="https://github.com/user-attachments/assets/cb10074e-11cf-49fe-88be-574e1edc9469" />

riomar0001 replied:
Hello Sean, 

Please confirm if this is what you mean in this request?

<img width="784" height="364" alt="image" src="https://github.com/user-attachments/assets/7fa6e2a6-d4c4-453f-b10e-2023f0bebb2f" />

seangaaab commented:
`frontend/src/pages/client/pycon/register/footer/useRegisterFooter.ts`

In line `244`, insert another `if` statement block that checks if `sprintDay` (From registration values. Utilize the same function in line `172`) is `true`. If so, check if the sprint day slots are full, and if it is already full, `setValue` of `sprintDay` to `false`.

riomar0001 committed:
fix: fix amplify dummy name

chore: change the indication for sold out in sprint selection card

riomar0001 committed:
chore: wrap onchange for sprintIsSoldOut in useEffect

riomar0001 commented:
> Quote reply
> Reference in

riomar0001 committed:
chore: add if statement on line 244 to check if sprintDay is full

seangaaab commented on frontend/src/pages/client/pycon/register/footer/useRegisterFooter.ts:
```diff
@@ -242,6 +242,25 @@ export const useRegisterFooter = (
+        return;
+      }
+
+      console.log(event.data);
```
remove console.log

seangaaab commented on frontend/src/pages/client/pycon/register/footer/useRegisterFooter.ts:
```diff
@@ -242,6 +242,25 @@ export const useRegisterFooter = (
+      const event = await api.execute(getEvent(eventId));
+
+      if (!event.data || event.status !== 200) {
+        return;
```
add toast notification to retry submitting

seangaaab replied:
@riomar0001  @jlorion Update error message to Try again instead.

jlorion replied:
@seangaaab

seangaaab commented on frontend/src/pages/client/pycon/register/footer/useRegisterFooter.ts:
```diff
@@ -242,6 +242,25 @@ export const useRegisterFooter = (
+        errorToast({
+          title: 'Sprint Day is full',
+          description: 'Sorry, but the Sprint Day slots are already full. Your registration will proceed without Sprint Day.'
+        });
```
after errorToast, insert a function that sets the step back to ticket selection step. You may refer the same function call in line 240

seangaaab commented on frontend/src/pages/client/pycon/register/steps/TicketSelectionStep.tsx:
```diff
@@ -226,14 +232,24 @@ const TicketType: FC<TicketTypeProps> = ({ ticketType, benefits, subtitle, value
     <div className="flex flex-col gap-6 mt-12">
       <div className="text-center md:text-left">
-        <div className="flex items-center gap-3 mb-2 justify-center md:justify-start">
+        <div className="flex items-c  enter gap-3 mb-2 justify-center md:justify-start">
```
Fix this class

riomar0001 committed:
fix: fix code in useRegisterFooter

- remove console log
- add errortoast in line 249 conditional statement
- setCurrentStep to stepTicker Selection when sprint days is full

ASPactores commented on backend/usecase/event_usecase.py:
```diff
@@ -95,6 +95,12 @@ def update_event(self, event_id: str, event_in: EventIn) -> Union[JSONResponse,
         original_status = original_event.status
 
+        if event_in.maximumSprintDaySlots:
+            registrations = self.__registration_repository.query_registrations(event_id)[1]
```
use tuple deconstruction for better readability (example: status, registrations, _ = ...).

jlorion committed:
fix(event): tuple deconstuction fix for readability

ASPactores reviewed (approved):
lgtm

jlorion committed:
fix: error message to try again

seangaaab reviewed (approved):
LGTM
